arm-network/2016-09-01/swagger/network.json

```diff
@@ -7975,10 +8048,15 @@
           "description": "Gets or sets resource guid property of the VirtualNetworkGatewayConnection resource"
         },
         "provisioningState": {
+          "readOnly": true,
           "type": "string",
           "description": "Gets provisioning state of the VirtualNetworkGatewayConnection resource Updating/Deleting/Failed"
         }
       },
+      "required": [
+        "virtualNetworkGateway1",
+        "connectionType"
+      ],
       "description": "VirtualNetworkGatewayConnection properties"
     },
     "VirtualNetworkGatewayConnection": {
```

Comment:
"properties" should be a required property in VirtualNetworkGatewayConnection because it's type "VirtualNetworkGatewayConnectionPropertiesFormat" has two required properties "virtualNetworkGateway1", and  "connectionType". Since "VirtualNetworkGatewayConnectionPropertiesFormat" is being flattened (`"x-ms-client-flatten":true`, it becomes even more important to mark "properties" property of "VirtualNetworkGatewayConnection" required.